[build] Check for Python 3 in configure.ac

monojenkins opened the pull request:
!! This PR is a copy of mono/mono#18625,  please do not edit or review it in this repo !!<br/>!! Merge the PR only after the original PR is merged !!<br/><hr/><br/>I fear this causing packaging problems, but as-is we require Python 3 for most of our Python files but don't actually check for that in configure. We should be consistent about this. Some projects (notably wasm) also use the Python version detected here, so bumping this will also bump the version used in those cases (a good thing, since there are some known bugs when using the scripts with Python 2).

CoffeeFlux commented:
Looks like our Linux build machines don't have Python 3 installed? That seems like a mistake. cc: @directhex @akoeplinger

directhex commented:
@CoffeeFlux file an issue on https://github.com/dotnet/dotnet-buildtools-prereqs-docker I guess. We use the same Centos image as CoreCLR

akoeplinger commented:
Python3 should be available everywhere since it's a Helix dependency, but maybe not  3.5+.

Can you try with a lower version, or do we really require 3.5?

CoffeeFlux commented:
I can swap to a lower version; I have no idea what we actually require without going through all our scripts. See: https://github.com/mono/mono/pull/18625#issuecomment-580010631

CoffeeFlux commented:
With that said, I'm somewhat hesitant to go much lower than 3.4 unless we have to. Python 3.4 is already EOL...

akoeplinger commented:
As long as it builds Mono I don't really mind if someone uses an EOL Python, so I'd start with 3.0 :)

directhex commented:
Python 3 is NOT a part of the image we build on x64 Linux, specified here: https://github.com/dotnet/runtime/blob/master/eng/pipelines/common/platform-matrix.yml#L146

```
directhex@breakfast:~/Projects/coreclr$ docker run -it cbcc06de207d bash
[root@0e99fec377fa /]# python
python            python-config     python2           python2-config    python2.7         python2.7-config  
```

akoeplinger commented:
Ah interesting, so that image is only used for building then and not on Helix. Let's get python3 into the images on https://github.com/dotnet/dotnet-buildtools-prereqs-docker then.

directhex commented:
I just checked, python3 is also missing from the Alpine image for our musl build, but is present on the Ubuntu image used for Wasm

CoffeeFlux committed:
[build] Check for Python 3 in configure.ac

I fear this causing packaging problems, but as-is we require Python 3 for most of our Python files but don't actually check for that in configure. We should be consistent about this. Some projects (notably wasm) also use the Python version detected here, so bumping this will also bump the version used in those cases (a good thing, since there are some known bugs when using the scripts with Python 2).

CoffeeFlux committed:
Update docker images to reflect py3 inclusion

akoeplinger committed:
Update centos-7 image

CoffeeFlux reviewed (approved):
Mono-side PR approved.
